tests/unit/test_cli.py

```diff
@@ -211,3 +214,63 @@ def testReferenceBasesListArguments(self):
         self.assertEqual(args.end, 2)
         self.assertEquals(args.outputFormat, "fasta")
         self.assertEquals(args.runner, cli.ListReferenceBasesRunner)
+
+
+class TestOutputFormats(unittest.TestCase):
+    """
+    Tests the different output formats of the cli
+    """
+    class FakeArgs(object):
+        def __init__(self, outputFormat='text'):
+            self.outputFormat = outputFormat
+            self.id = 'id'
+            self.key = 'key'
+            self.baseUrl = 'baseUrl'
+            self.verbose = 'verbose'
+
+    class FakeObject(protocol.ProtocolElement):
+
+        __slots__ = ['id', 'name']
+
+        def __init__(self):
+            self.id = 'id'
+            self.name = 'name'
+
+    def _getRunPrintMethodCalls(self, runner):
+        printCalls = []
+        with mock.patch('__builtin__.print') as printMethod:
+            printMethod.side_effect = \
+                lambda *args, **kwargs: printCalls.append((args, kwargs))
+            runner.run()
+        return printCalls
+
+    def testListReferenceBasesFasta(self):
+        args = self.FakeArgs('fasta')
```

Comment:
OK, point (1) is well taken, and convincing.  We also need integrationy test for this though--- we should test the end-to-end correctness of FASTA output.

Point (2) isn't true any more though: we now have some rather snazzy machinery for running CLI commands and capturing the output, without forking a subprocess. See ``tests/end_to_end/test_client_json.py`` for examples.